python/src/main.cpp

```diff
@@ -32,15 +32,24 @@ PYBIND11_MODULE(kp, m) {
         .value("storage", kp::Tensor::TensorTypes::eStorage, "Tensor with host visible gpu memory.")
         .export_values();
 
+
+
+
+
     py::class_<kp::Tensor, std::shared_ptr<kp::Tensor>>(m, "Tensor", DOC(kp, Tensor))
         .def(py::init(
-            [](const std::vector<float>& data) {
-                return std::unique_ptr<kp::Tensor>(new kp::Tensor(data));
-            }), DOC(kp, Tensor, Tensor, 2))
-        .def(py::init(
-            [](const std::vector<float>& data, kp::Tensor::TensorTypes tensorTypes) {
-                return std::unique_ptr<kp::Tensor>(new kp::Tensor(data, tensorTypes));
-            }), "Initialiser with list of data components and tensor GPU memory type.")
+            [np](const py::array_t<float> data, kp::Tensor::TensorTypes tensor_type) {
```

Comment:
It seems that the tensor_type would require a default value in order to be equivallent, or would this also allow for only data to be passed?

Reply:
Same here, `tensor_type` is made optional in line 51: `py::arg("tensor_type") = kp::Tensor::TensorTypes::eDevice`

Reply:
Oh I see, I didn't know you could specify the arguments and their values as extra parameters in the `.def(...)` function, ok makes complete sense, looks good